File: maths/algebre/factorielle/Factorielle.java
// -----------------------------------------------
// Fonction factorielle
// -----------------------------------------------
// Calcul du produit des entiers de 1 a n, où n un
// entier positif est passe en argument.
// -----------------------------------------------

public class Factorielle {

// -----------------------------------------------
// Version courte
// -----------------------------------------------
// Cette version permet de comprendre le contexte
// de la fonction et sa logique.

public static int factorielle(int nombre) {
    int produit = 1;
    
    for (int i = 1; i <= nombre; i++) {
        produit *= i;
    }
    
    return produit;
}

//-----------------------------------------------
// Application
//-----------------------------------------------

public static void main(String[] args) {

    int res = factorielle(6);
    System.out.print("La factorielle de 6");
    System.out.println("est " + res);
}

// -----------------------------------------------
// Version golf
// -----------------------------------------------
// Version condensee et optimisee du code, utili-
// sant des noms de variables courts et combinant
// certaines operations pour reduire la taille du
// code. Pour la beaute du geste !

public static int F(int n){
int p=1;for(int i=1;i<=n;i++)p*=i;return p;}

//-----------------------------------------------
// Version commentee
//-----------------------------------------------
// Proche de la version raccourcie, mais avec des
// commentaires concis pour expliquer les etapes
// principales de la fonction.

public static int factorielle_commentee(
    int nombre) {

    // Initialiser le produit a 1
    int produit = 1;
    
    // De 1 a nombre,
    for (int i = 1; i <= nombre; i++) {

        // multiplier par le nombre courant.
        produit *= i;
    
    }
    
    // Retourner la factorielle calculee.
    return produit;
}
}









//

File: algebre/factorielle/factorielle.py

# -----------------------------------------------
# Fonction factorielle
# -----------------------------------------------
# Calcul du produit des entiers de 1 a n, où n un
# entier positif est passe en argument.
# -----------------------------------------------


# -----------------------------------------------
# Version courte
# -----------------------------------------------

def factorielle(nombre):
    produit = 1
 
    for nombre_actuel in range(1, nombre + 1):
        produit *= nombre_actuel
 
    return produit



# -----------------------------------------------
# Application
# -----------------------------------------------

resultat = factorielle(6) 
print(f"La factorielle de 6 est : {resultat}")



# -----------------------------------------------
# Version golf
# -----------------------------------------------

F=lambda n: (lambda f:f(f,n))(lambda s,x:1 if x==
    0 else x*s(s,x-1))



# -----------------------------------------------
# Version commentee
# -----------------------------------------------


def factorielle_commentee(nombre):
 
 # Initialiser du produit a 1.
 produit = 1
 
 # De 1 a nombre,
 for nombre_actuel in range(1, nombre + 1):

  # Multiplier par le nombre courant.
  produit *= nombre_actuel
 
 # Retourner la factorielle calculee.
 return produit









#